api/routes/analysis.py

```diff
@@ -309,3 +320,67 @@ async def get_sample_reads_map_for_seqr(
         media_type='text/tab-separated-values',
         headers={'Content-Disposition': f'filename={basefn}.tsv'},
     )
+
+
+@router.post('/sample-file-sizes', operation_id='getSampleFileSizes')
+async def get_sample_file_sizes(
+    start_date: str = None,
+    end_date: str = None,
+    project_names: List[str] = None,
+    connection: Connection = get_projectless_db_connection,
+) -> list[ProjectSizeModel]:
+    """
+    Get the per sample file size by type over the given projects and date range
+    """
+    atable = AnalysisLayer(connection)
+    stable = SampleLayer(connection)
+
+    # Check access to projects
+    project_ids = None
+    if project_names:
+        pt = ProjectPermissionsTable(connection=connection.connection)
+        project_ids = await pt.get_project_ids_from_names_and_user(
+            connection.author, project_names, readonly=True
+        )
+
+    prj_name_map = dict(zip(project_ids, project_names))
+
+    # Try to convert dates
+    start, end = None, None
+    if start_date or end_date:
+        try:
+            if start_date:
+                start = datetime.strptime(start_date, '%Y-%m-%d').date()
+
+            if end_date:
+                end = datetime.strptime(end_date, '%Y-%m-%d').date()
+        except Exception as excep:
+            raise ValueError(
+                f'Start or end date could not be converted: {start_date} {end_date}'
+            ) from excep
+
+    # Get samples from pids
+    samples = await stable.get_samples_by(project_ids=project_ids)
```

Comment:
This logic should be moved to the AnalysisLayer so it's potentially reusable.

Reply:
If the only result is just getting active sample_ids (as internal ints), we should add a function to do that to avoid `int -> Sample.id: str -> int`.